docs/HowToCertificatesRotation.md

```diff
@@ -0,0 +1,486 @@
+
+- `processCertificates` - information about certificates used by each database process identified by `startId`.
+- `processTrusted` - trusted certificate aliases (referenced in `trustedCertificates` section) for each database process identified by `startId`.
+- `serverCertificates` - information about certificates used by each AP identified by `serverId`.
+- `serverTrusted` - trusted certificate aliases (referenced in `trustedCertificates` section) for each AP identified by `serverId`.
+- `trustedCertificates` - information about trusted certificates in the domain identified by `alias`.
+
+Example output:
+
+```json
+{
+  "processCertificates": {
+    "11": {
+      "caPathLength": -1,
+      "certificatePem": "-----BEGIN CERTIFICATE-----...-----END CERTIFICATE-----",
+      "expires": 1659087997000,
+      "expiresTimestamp": "2022-07-29T09:46:37.000+0000",
+      "issuerName": "CN=*.nuodb.svc.cluster.local, OU=Eng, O=NuoDB, L=Boston, ST=MA, C=US, SERIALNUMBER=67890",
+      "subjectName": "CN=172.17.0.8"
+    },
+    "12": {
+      "caPathLength": -1,
+      "certificatePem": "-----BEGIN CERTIFICATE-----...-----END CERTIFICATE-----",
+      "expires": 1659087998000,
+      "expiresTimestamp": "2022-07-29T09:46:38.000+0000",
+      "issuerName": "CN=*.nuodb.svc.cluster.local, OU=Eng, O=NuoDB, L=Boston, ST=MA, C=US, SERIALNUMBER=67890",
+      "subjectName": "CN=172.17.0.9"
+    },
+    "9": {
+      "caPathLength": -1,
+      "certificatePem": "-----BEGIN CERTIFICATE-----...-----END CERTIFICATE-----",
+      "expires": 1659087974000,
+      "expiresTimestamp": "2022-07-29T09:46:14.000+0000",
+      "issuerName": "CN=*.nuodb.svc.cluster.local, OU=Eng, O=NuoDB, L=Boston, ST=MA, C=US, SERIALNUMBER=67890",
+      "subjectName": "CN=172.17.0.5"
+    }
+  },
+  "processTrusted": {
+    "11": [
+      "ca_prime",
+      "nuocmd_prime"
+    ],
+    "12": [
+      "ca_prime",
+      "nuocmd_prime"
+    ],
+    "9": [
+      "ca_prime",
+      "nuocmd_prime"
+    ]
+  },
+  "serverCertificates": {
+    "admin-nuodb-cluster0-0": {
+      "caPathLength": 2147483647,
+      "certificatePem": "-----BEGIN CERTIFICATE-----...-----END CERTIFICATE-----",
+      "expires": 4781150383000,
+      "expiresTimestamp": "2121-07-05T09:19:43.000+0000",
+      "issuerName": "CN=ca.nuodb.com, OU=Eng, O=NuoDB, L=Boston, ST=MA, C=US, SERIALNUMBER=67890",
+      "subjectName": "CN=*.nuodb.svc.cluster.local, OU=Eng, O=NuoDB, L=Boston, ST=MA, C=US, SERIALNUMBER=67890"
+    },
+    "admin-nuodb-cluster0-1": {
+      "caPathLength": 2147483647,
+      "certificatePem": "-----BEGIN CERTIFICATE-----...-----END CERTIFICATE-----",
+      "expires": 4781150383000,
+      "expiresTimestamp": "2121-07-05T09:19:43.000+0000",
+      "issuerName": "CN=ca.nuodb.com, OU=Eng, O=NuoDB, L=Boston, ST=MA, C=US, SERIALNUMBER=67890",
+      "subjectName": "CN=*.nuodb.svc.cluster.local, OU=Eng, O=NuoDB, L=Boston, ST=MA, C=US, SERIALNUMBER=67890"
```

Comment:
The serial number is actually the same for all certificates (DB process, AP, CA)? I would expect this to be unique for each certificate generated.

Reply:
This is an output from real deployment. It looks like I used the same `SERIALNUMBER` for the AP, CA, and client keys. The DB process doesn't have it as it is generated by the admin which probably put the engine address as CN.

I can randomize the `SERIALNUMBER` values for different keys so that there is no confusion if the same one should be used or not.